### PR TITLE
Add dockerhub readonly user login before gradle check

### LINKED ISSUE #3358: [Bug]: Gradle check workflows are failing with docker throttling
### Describe the bug

As far as I can tell, all gradle check workflows in the opensearch-project/OpenSearch repository are failing with some version of the following error:

```
> Task :test:fixtures:gcs-fixture:composeBuild FAILED
Building gcs-fixture
Service 'gcs-fixture' failed to build: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```

https://build.ci.opensearch.org/job/gradle-check/13164/ is the first instance I have found and it ran at 6:17:46 PM (UTC)

### To reproduce

Do anything to trigger a gradle check workflow (e.g. submit a PR)

### Expected behavior

_No response_

### Host / Environment

_No response_

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Add dockerhub readonly user login before gradle check

### Issues Resolved
#3358

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: jenkins/gradle/gradle-check.jenkinsfile [hunk 1]
```diff
@@ -112,10 +112,15 @@ pipeline {
                         def repo_url = "${GIT_REPO_URL}".replace(".git", "/commit")
                         currentBuild.description = """runner: ${agent_name}<br>git: <a href="${GIT_REPO_URL}">${GIT_REPO_URL}</a><br>ref: <a href="${repo_url}/${GIT_REFERENCE}">${GIT_REFERENCE}</a>"""
 
-                        runGradleCheck(
-                            gitRepoUrl: "${GIT_REPO_URL}",
-                            gitReference: "${GIT_REFERENCE}"
-                        )
+                        def CREDENTIAL_ID = 'jenkins-production-dockerhub-credential-readonly'
```
Comment (gaiksaya) [inline jenkins/gradle/gradle-check.jenkinsfile:115]: It's a production user?

**Review by gaiksaya — APPROVED** [review]

**Comment by gaiksaya** [comment]
Need to move gradle check file to OpenSearch core repo and add tests for the same.

**Comment by peterzhuamazon** [comment]
Need to hold on this PR before others get merged.

**Review by zelinh — DISMISSED** [review]
Block until other PRs merged.

**Pull request merged**